variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactory.java

```diff
@@ -80,7 +80,7 @@
         for (int altAlleleIdx = 0; altAlleleIdx < alternateAlleles.length; altAlleleIdx++) {
             VariantCoreFields keyFields = generatedKeyFields.get(altAlleleIdx);
             Variant variant = new Variant(chromosome, keyFields.getStart(), keyFields.getEnd(), keyFields.getReference(),
-                                          keyFields.getAlternate());
+                                          keyFields.getAlternate(), null);
```

Comment:
I wonder if we should have 2 constructors, one with and another without the `mainId` parameter. The one without it will be used by the pipeline that creates new variants, ensuring that we don't assign the ID even by mistake. The one with it will be used by the web services. We could document them to make clear when each of them should be used.